validator/remote/src/test/java/tech/pegasys/teku/validator/remote/RemoteValidatorApiHandlerTest.java

```diff
@@ -0,0 +1,347 @@
+  }
+
+  @Test
+  public void sendSAggregateAndProof_InvokeApiWithCorrectRequest() {
+    final AggregateAndProof aggregateAndProof = dataStructureUtil.randomAggregateAndProof();
+    final BLSSignature signature = dataStructureUtil.randomSignature();
+    final SignedAggregateAndProof signedAggregateAndProof =
+        new SignedAggregateAndProof(aggregateAndProof, signature);
+
+    tech.pegasys.teku.api.schema.SignedAggregateAndProof schemaSignedAggAndProof =
+        new tech.pegasys.teku.api.schema.SignedAggregateAndProof(signedAggregateAndProof);
+
+    ArgumentCaptor<tech.pegasys.teku.api.schema.SignedAggregateAndProof> argumentCaptor =
+        ArgumentCaptor.forClass(tech.pegasys.teku.api.schema.SignedAggregateAndProof.class);
+
+    apiHandler.sendAggregateAndProof(signedAggregateAndProof);
+
+    verify(apiClient).sendAggregateAndProof(argumentCaptor.capture());
+    assertThat(argumentCaptor.getValue())
+        .usingRecursiveComparison()
+        .isEqualTo(schemaSignedAggAndProof);
+  }
+
+  @Test
+  public void subscribeToBeaconCommitteeForAggregation_InvokeApi() {
+    final int committeeIndex = 1;
+    final UInt64 aggregationSlot = UInt64.ONE;
+
+    apiHandler.subscribeToBeaconCommitteeForAggregation(committeeIndex, aggregationSlot);
+
+    verify(apiClient)
+        .subscribeToBeaconCommitteeForAggregation(eq(committeeIndex), eq(aggregationSlot));
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void subscribeToPersistentSubnets_InvokeApi() {
+    final int subnetId = 1;
+    final UInt64 slot = UInt64.ONE;
+
+    final SubnetSubscription subnetSubscription = new SubnetSubscription(subnetId, slot);
+    final tech.pegasys.teku.api.schema.SubnetSubscription schemaSubnetSubscription =
+        new tech.pegasys.teku.api.schema.SubnetSubscription(subnetId, slot);
+
+    ArgumentCaptor<Set<tech.pegasys.teku.api.schema.SubnetSubscription>> argumentCaptor =
+        ArgumentCaptor.forClass(Set.class);
+
+    apiHandler.subscribeToPersistentSubnets(Set.of(subnetSubscription));
+
+    verify(apiClient).subscribeToPersistentSubnets(argumentCaptor.capture());
+
+    final Set<tech.pegasys.teku.api.schema.SubnetSubscription> request = argumentCaptor.getValue();
+    assertThat(request).hasSize(1);
+    assertThat(request.stream().findFirst().get())
+        .usingRecursiveComparison()
+        .isEqualTo(schemaSubnetSubscription);
+  }
+
+  private <T> Optional<T> unwrapToOptional(SafeFuture<Optional<T>> future) {
+    try {
+      return future.get();
+    } catch (Exception e) {
+      fail("Error unwrapping optional from SafeFuture", e);
+      throw new RuntimeException(e);
+    }
+  }
+
+  private <T> T unwrapToValue(SafeFuture<Optional<T>> future) {
+    try {
+      return future.get().orElseThrow();
+    } catch (Exception e) {
+      fail("Error unwrapping value from SafeFuture", e);
+      throw new RuntimeException(e);
+    }
+  }
```

Comment:
Probably better to use `Waiter.waitFor` here so it will timeout. Or if you move to using `AsyncRunner` you could pass in a `StubAsyncRunner` and then use `StubAsyncRunner.executeQueuedTasks()` to trigger execution and the whole test would wind up being single threaded.